playwright-junit/src/main/java/com/microsoft/playwright/ExtensionUtils.java

```diff
@@ -0,0 +1,28 @@
+package com.microsoft.playwright;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.util.Locale;
+
+import static org.junit.platform.commons.support.AnnotationSupport.findAnnotation;
+import static org.junit.platform.commons.support.AnnotationSupport.isAnnotated;
+
+class ExtensionUtils {
+  // Necessary because of https://github.com/junit-team/junit5/issues/3447
+  static ExecutionMode getExecutionMode(ExtensionContext extensionContext) {
+    return findAnnotation(extensionContext.getTestMethod(), Execution.class)
```

Comment:
We would still need this method because of the bug I referenced unfortunately. 

Supporting per method is not adding any extra complexity aside from knowing when to close the objects. 

The creation of all objects, regardless of happens when the ParameterResolver runs and that would be up to the user whether they need it in a hook or the test method.